auction-server/src/opportunity/api.rs

```diff
@@ -729,7 +736,10 @@ pub async fn get_opportunities(
         Ok(Json(
             opportunities
                 .into_iter()
-                .take(repository::OPPORTUNITY_PAGE_SIZE as usize)
+                .take(std::cmp::min(
+                    query_params.limit,
+                    OPPORTUNITY_PAGE_SIZE_CAP as usize,
```

Comment:
i think it's better to return the max rather than error out. maybe we could add a note/warning field to the message, but i think it's bad to error here since the user can still recover some data and go on to recover more data by paginating

Reply:
people are not going to read the docs carefully and will use a large number here. Let's say a thousand. Then they will see that the endpoint is returning them 44 values and think ok, this would work later on and can return 1000 values. This would lead to a silent failure and unnoticed bugs but if we validate everything in the beginning and throw errors they will catch it in testing phase and wouldn't have any issues on prod.